### PR TITLE
adds error handling for get-data function

### LINKED ISSUE #307: Feedback mechanism for client side errors
using https://github.com/joostlawerman/SnackbarLightjs

### PR DESCRIPTION
**Pull request by Danwhy**
Closes #307

### CODE REVIEW

**Comment by alanshaw** [comment]
👓

File: client/lib/archive-listeners.js [hunk 1]
```diff
@@ -39,7 +40,11 @@ module.exports = (ctx, listeners) => {
       headers: { Accept: 'application/vnd.api+json' }
     };
 
-    getData(url, opts, function (json) {
+    getData(url, opts, function (err, json) {
+      if (err) {
+        console.warn(err);
+        Snackbar.create('Error getting data from the server');
```
Comment (alanshaw) [inline client/lib/archive-listeners.js]: Need to return here in all of these instances.

Comment (alanshaw) [inline client/lib/archive-listeners.js]: `console.error` is more appropriate here

File: client/lib/archive-listeners.js [hunk 2]
```diff
@@ -39,7 +40,11 @@ module.exports = (ctx, listeners) => {
       headers: { Accept: 'application/vnd.api+json' }
     };
 
-    getData(url, opts, function (json) {
+    getData(url, opts, function (err, json) {
+      if (err) {
+        console.warn(err);
```
Comment (alanshaw) [inline client/lib/archive-listeners.js]: `console.error` is more appropriate here

**Comment by alanshaw** [comment]
@Danwhy just a couple of minor issues but this is good!

**Comment by Danwhy** [comment]
@alanshaw :+1: Should be good to go now

**Comment by alanshaw** [comment]
I think you missed the first one I commented on in `client/lib/archive-listeners.js`

**Comment by Danwhy** [comment]
oops, fixed now

**Pull request merged**